examples/controlnet/train_controlnet_sd3.py

```diff
@@ -1094,7 +1094,7 @@ def load_model_hook(models, input_dir):
         weight_dtype = torch.bfloat16
 
     # Move vae, transformer and text_encoder to device and cast to weight_dtype
-    vae.to(accelerator.device, dtype=torch.float32)
+    vae.to(accelerator.device, dtype=weight_dtype)
```

Comment:
I think it should be like:

```python
if args.upcast_vae:
    vae.to(accelerator.device, dtype=torch.float32)
else:
    vae.to(accelerator.device, dtype=weight_dtype)
```

With `upcast_vae` being added to the arguments?